examples/cpp/Schrodinger2D.cpp

```diff
@@ -0,0 +1,156 @@
+#include "mole.h"    
+#include <iostream>    
+#include <cmath>    
+#include <iomanip>    
+#include <vector>    
+#include <fstream>    
+  
+using namespace arma;    
+  
+int main() {    
+  int p = 105;  // Number of time steps for the simulation    
+  double Lxy = 1.0;    
+  int k = 2;  // Order of accuracy    
+  int m = 50; // Grid points in x    
+  int n = 50; // Grid points in y    
+  int nx = 2; // Energy level in x    
+  int ny = 2; // Energy level in y    
+  double dx = Lxy / m; // Step in x    
+  double dy = Lxy / n; // Step in y    
+  double dt = dx/2; // Reduced time step size with additional reduction    
+  
+  // Define staggered grids    
+  vec xgrid = join_vert(vec({0}), linspace(dx / 2, Lxy - dx / 2, m), vec({Lxy}));    
+  vec ygrid = join_vert(vec({0}), linspace(dy / 2, Lxy - dy / 2, n), vec({Lxy}));    
+  
+  // Initialize 2D staggered grid    
+  mat X, Y;    
+  Utils utils;    
+  utils.meshgrid(xgrid, ygrid, X, Y);    
+  
+  // Debug grid sizes    
+  std::cout << "Grid X size: " << X.n_rows << " x " << X.n_cols << std::endl;    
+  std::cout << "Grid Y size: " << Y.n_rows << " x " << Y.n_cols << std::endl;    
+  
+  // Initialize Laplacian operator with Robin BC    
+  Laplacian L(k, m, n, dx, dy);    
+  RobinBC BC(k, m, 1, n, 1, 1, 0);    
+  L = L + BC;    
+  
+  // Ensure the Laplacian is square    
+  int total_size = (m + 2) * (n + 2); // Total size for the grid including boundaries    

```

Comment:
the vector v_grid does not have this size, it is bigger. Check the matlab code for when it is initialized.